### PR TITLE
Zarr

### PR DESCRIPTION
**Pull request by Thechopsee**
Cleaned version of PR

### CODE REVIEW

File: src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceEndpoint.java [hunk 1]
```diff
@@ -152,26 +153,59 @@ public Response startDatasetServer(@PathParam(UUID) String uuid,
 		}
 	}
 
+	@GET
+	@Path("datasets/{" + UUID + "}/type")
```
Comment (kozusznik) [inline src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceEndpoint.java]: Is it neccesary to add new resource to REST API. Type could be returned in DatasetDTO in query
GET /datasets/{UUID}

File: src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceEndpoint.java [hunk 2]
```diff
@@ -152,26 +153,59 @@ public Response startDatasetServer(@PathParam(UUID) String uuid,
...
+
+	}
+
+
 	@POST
 	@Path("datasets/")
 	@Consumes(MediaType.APPLICATION_JSON)
 	public Response createEmptyDataset(DatasetDTO dataset)
 	{
+		String test=dataset.getCompression();
```
Comment (kozusznik) [inline src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceEndpoint.java]: I suggest to NOT pass type in compression. I guess that more intuitive is attribute type in an object of DatasetDTO class.

File: src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceEndpoint.java [hunk 3]
```diff
@@ -152,26 +153,59 @@ public Response startDatasetServer(@PathParam(UUID) String uuid,
...
+		log.info(splitted[0]);
+		//if(test.equals(gzip))
 		log.info("creating empty dataset");
 		log.debug("dataset=" + dataset);
-		try {
-			java.util.UUID result = datasetRegisterServiceImpl.createEmptyDataset(
-				dataset);
-			return Response.ok().entity(result.toString()).type(
-				MediaType.TEXT_PLAIN).build();
+		if(splitted.length==1)
```
Comment (kozusznik) [inline src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceEndpoint.java]: These two blocks are same - they could be parametrized with a type of output file ("N5" or something read from data.)

File: src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceEndpoint.java [hunk 4]
```diff
@@ -152,26 +153,59 @@ public Response startDatasetServer(@PathParam(UUID) String uuid,
...
+						MediaType.TEXT_PLAIN).build();
+			}
+			catch (Exception exc) {
+				log.warn("read", exc);
+				return Response.serverError().entity(exc.getMessage()).type(
+						MediaType.TEXT_PLAIN).build();
+			}
 		}
 	}
-
```
Comment (kozusznik) [inline src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceEndpoint.java:174]: Consider leaving a empty line and remove more then two empty lines between methods.

File: src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceImpl.java [hunk 5]
```diff
@@ -138,18 +140,19 @@ public void addExistingDataset(String uuid) throws IOException,
 		}
 	}
```
Comment (kozusznik) [inline src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceImpl.java]: Consider pass datasettype in a DTO.

File: src/main/java/cz/it4i/fiji/datastore/zarr/fileTypeEnum.java
Comment (kozusznik) [inline src/main/java/cz/it4i/fiji/datastore/zarr/fileTypeEnum.java:1]: Name of enum identifier should start with upper case letter according to Java code convention.

File: src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceImpl.java [hunk 6]
```diff
@@ -222,6 +225,19 @@ public String getCommonMetadata(String uuid) {
 		Dataset dataset = getDataset(uuid);
 		return Strings.nullToEmpty(dataset.getMetadata());
 	}
+	public fileTypeEnum getDataType(String uuid)
+	{
+		Dataset dataset = getDataset(uuid);
+		String type=dataset.getDatasetType();
+		if(type.equals("Zarr"))
```
Comment (kozusznik) [inline src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRegisterServiceImpl.java]: use Objects.equals - it guess that type should be null.

File: src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRepository.java [hunk 7]
```diff
@@ -44,7 +44,25 @@ public class DatasetRepository implements PanacheRepository<Dataset>,
 
 	@Inject
 	ApplicationConfiguration configuration;
+	public String findTypebyUUID(String uuid)
```
Comment (kozusznik) [inline src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRepository.java:47]: Is it necessary as the type is part of a dataset?

File: src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRepository.java [hunk 8]
```diff
@@ -44,7 +44,25 @@ public class DatasetRepository implements PanacheRepository<Dataset>,
...
+						uuid)).singleResultOptional();
+		if (result.isEmpty()) {
+			throw new NotFoundException("Dataset with UUID = " + uuid +
+					" not found ");
+		}
+		Dataset resultDataset = result.get();
 
+		return resultDataset.getDatasetType();
+	}
+	public void updateDatasetTypebyUUID(String uuid, String newDatasetType) {
```
Comment (kozusznik) [inline src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRepository.java:60]: Is it necessary as the type is part of a dataset?

**Review by kozusznik — CHANGES_REQUESTED** [review]
Create Unit test and make experiments measuring performance.

File: src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRepository.java (hunk 7 above)
Comment (sebastianwalenta) [reply src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRepository.java:47]: Probably yes, I don't know how else I would solve the choice of the right handler for writing/reading datasets.

**Comment by sebastianwalenta** [comment]
I hope the rebase went well and the commits are as they should be.

**Pull request merged**